### PR TITLE
MODDATAIMP-491: Improve logging to be able to trace the path of each record and file_chunks.

### PR DESCRIPTION
**Pull request by VRohach**
Fix OCLC issue (NPE fixed).

### CODE REVIEW

**Review by RuslanLavrov — APPROVED** [review]

File: src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java [hunk 1]
```diff
@@ -72,12 +74,19 @@ public KafkaEventPublisher(KafkaConfig kafkaConfig, Vertx vertx, int maxDistribu
...
-        KafkaHeader.header(OKAPI_URL_HEADER, eventPayload.getOkapiUrl()),
-        KafkaHeader.header(OKAPI_TENANT_HEADER, eventPayload.getTenant()),
-        KafkaHeader.header(OKAPI_TOKEN_HEADER, eventPayload.getToken()),
-        KafkaHeader.header(RECORD_ID_HEADER, recordId),
-        KafkaHeader.header(CHUNK_ID_HEADER, chunkId)));
+      List<KafkaHeader> headers = new ArrayList<>();
+      headers.add(KafkaHeader.header(OKAPI_URL_HEADER, eventPayload.getOkapiUrl()));
+      headers.add(KafkaHeader.header(OKAPI_TENANT_HEADER, eventPayload.getTenant()));
+      headers.add(KafkaHeader.header(OKAPI_TOKEN_HEADER, eventPayload.getToken()));
+      if (recordId != null) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java:81]: Better to check using StringUtils.isNotBlank()

File: src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java [hunk 2]
```diff
@@ -72,12 +74,19 @@ public KafkaEventPublisher(KafkaConfig kafkaConfig, Vertx vertx, int maxDistribu
...
+        LOGGER.warn("RecordId is empty for jobExecutionId: '{}' ", jobExecutionId);
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java:82]: RecordId is NOT empty in this case

File: src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java (hunk 1 above)
Comment (VRohach) [reply src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java:81]: @SerhiiNosko  agree, thank you!

File: src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java (hunk 2 above)
Comment (VRohach) [reply src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java:82]: @ Yes! My bad, I will fix it.

File: src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java [hunk 3]
```diff
@@ -100,4 +103,17 @@ public KafkaEventPublisher(KafkaConfig kafkaConfig, Vertx vertx, int maxDistribu
...
   }
+
+  private void checkAndAddHeaders(String recordId, String chunkId, String jobExecutionId, List<KafkaHeader> headers) {
+    if (recordId == null) {
+      LOGGER.warn("RecordId is empty for jobExecutionId: '{}' ", jobExecutionId);
+    } else {
+      headers.add(KafkaHeader.header(RECORD_ID_HEADER, recordId));
+    }
+    if (chunkId == null) {
+      LOGGER.warn("ChunkId is empty for jobExecutionId: '{}' ", chunkId);
```
Comment (AivarIUsupov) [inline src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java:114]: There should be jobExecutionId instead of chunkId

Comment (VRohach) [reply src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java:114]: @AivarIUsupov thanks!

**Review by AivarIUsupov — APPROVED** [review]

**Review by SerhiiNosko — APPROVED** [review]

**Pull request merged**